src/platform/Ameba/FactoryDataProvider.cpp

```diff
@@ -418,6 +418,31 @@ CHIP_ERROR FactoryDataProvider::SetSetupPasscode(uint32_t setupPasscode)
     return CHIP_ERROR_NOT_IMPLEMENTED;
 }
 
+CHIP_ERROR FactoryDataProvider::GetDeviceName(MutableCharSpan & deviceNameSpan)
+{
+    CHIP_ERROR err = CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND;
+
+    if (kReadFromFlash)
+    {
+        ReturnErrorCodeIf(deviceNameSpan.size() < mFactoryData.dii.device_name.len, CHIP_ERROR_BUFFER_TOO_SMALL);
+        ReturnErrorCodeIf(!mFactoryData.dii.device_name.value, CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND);
+        memcpy(deviceNameSpan.data(), mFactoryData.dii.device_name.value, mFactoryData.dii.device_name.len);
```

Comment:
Where is this setting the size of deviceNameSpan?

Reply:
removed this functionality